File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/welcome_wizard/WelcomeWizard.java
package com.crpr.androidcinema.domain.welcome_wizard;

import android.app.Activity;

import com.crpr.androidcinema.presentation.common.AppView;
import com.crpr.androidcinema.presentation.common.IPresenter;
import com.crpr.androidcinema.presentation.welcome_wizard.fragments.DefaultSlideFragment;

import java.util.List;

import rx.Observable;

/**
 * Created by claudioribeiro on 09/07/16.
 */
public interface WelcomeWizard {

    interface View extends AppView {
        void buildSlides(List<DefaultSlideFragment> slides);
        void userIsDone();
    }

    interface Presenter extends IPresenter {
        void onCreate();
        Observable<List<DefaultSlideFragment>> getSlides();
        void userIsDoneWithWelcomeWizard();
    }

    interface Interactor {
        Observable<Boolean> updateWelcomeWizardDone(boolean isDone);
    }

    interface Process {
        Observable<Boolean> updateWelcomeWizardDone(boolean isDone);
    }

    interface Navigator {
        void navigate(Activity activity);
    }
}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/domain/welcome_wizard/WelcomeWizard.kt
package com.crpr.androidcinema.domain.welcome_wizard

import android.app.Activity

import com.crpr.androidcinema.presentation.common.Base
import com.crpr.androidcinema.presentation.welcome_wizard.fragments.DefaultSlideFragment

import rx.Observable

/**
 * Created by claudioribeiro on 09/07/16.
 */
interface WelcomeWizard {

    interface View : Base.View {
        fun buildSlides(slides: List<DefaultSlideFragment>)
        fun userIsDone()
    }

    interface Presenter : Base.Presenter {
        fun onCreate()
        val slides: Observable<List<DefaultSlideFragment>>
        fun userIsDoneWithWelcomeWizard()
    }

    interface Interactor {
        fun updateWelcomeWizardDone(isDone: Boolean): Observable<Boolean>
    }

    interface Process {
        fun updateWelcomeWizardDone(isDone: Boolean): Observable<Boolean>
        fun checkWelcomeWizardDone(): Observable<WelcomeWizardResult>
    }

    interface Navigator {
        fun navigate(activity: Activity)
    }
}
